clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -632,9 +635,11 @@ public void onFailure(RuntimeException e) {
         final RequestFuture<ListOffsetResult> listOffsetRequestsFuture = new RequestFuture<>();
         final Map<TopicPartition, OffsetData> fetchedTimestampOffsets = new HashMap<>();
         final Set<TopicPartition> partitionsToRetry = new HashSet<>();
+        final Set<TopicPartition> partitionsRequireMetadataUpdate = new HashSet<>(timestampsToSearch.keySet());
```

Comment:
Another idea might be to pass `partitionsToRetry` into `groupListOffsetRequests`.